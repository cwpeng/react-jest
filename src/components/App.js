import React from "react";
class App extends React.Component{
	constructor(props){
		super(props);
	}
	render(){
		let negativeList;
		if(this.props.data===null){
			negativeList=<li>No Data</li>;
		}else{
			negativeList=this.props.data.filter((number)=>{
				return number<0;
			}).map((number)=>{
				return <li>{number}</li>;
			});
		}
		return <>
			<h3>Unit Testing</h3>
			<main>Jest Snapshot Testing</main>
			<ul>{negativeList}</ul>
		</>;
	}
}
export default App;