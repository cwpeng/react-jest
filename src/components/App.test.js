import App from "./App.js";
import React from "react";
import renderer from "react-test-renderer";

test("Test App Component", ()=>{
	let tree=renderer.create(<App data={[3,-2,5,-1]} />).toJSON();
	expect(tree).toMatchSnapshot();

	tree=renderer.create(<App data={[3,-2,0,-1]} />).toJSON();
	expect(tree).toMatchSnapshot();

	tree=renderer.create(<App data={[]} />).toJSON();
	expect(tree).toMatchSnapshot();

	tree=renderer.create(<App data={null} />).toJSON();
	expect(tree).toMatchSnapshot();
});