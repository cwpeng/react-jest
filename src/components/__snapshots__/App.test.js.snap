// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test App Component 1`] = `
Array [
  <h3>
    Unit Testing
  </h3>,
  <main>
    Jest Snapshot Testing
  </main>,
  <ul>
    <li>
      -2
    </li>
    <li>
      -1
    </li>
  </ul>,
]
`;

exports[`Test App Component 2`] = `
Array [
  <h3>
    Unit Testing
  </h3>,
  <main>
    Jest Snapshot Testing
  </main>,
  <ul>
    <li>
      -2
    </li>
    <li>
      -1
    </li>
  </ul>,
]
`;

exports[`Test App Component 3`] = `
Array [
  <h3>
    Unit Testing
  </h3>,
  <main>
    Jest Snapshot Testing
  </main>,
  <ul />,
]
`;

exports[`Test App Component 4`] = `
Array [
  <h3>
    Unit Testing
  </h3>,
  <main>
    Jest Snapshot Testing
  </main>,
  <ul>
    <li>
      No Data
    </li>
  </ul>,
]
`;
